pkg/config/config.go

```diff
@@ -40,112 +59,112 @@ const (
-	)
+type config struct {
+	// viper instance for holding user provided configuration
+	user *viper.Viper
+	// viper instance for holding system specific configuration
+	system *viper.Viper
+	// the default configuration values to initialize with
+	defaultCfg types.BacalhauConfig
 }
 
-func Load(path string) (types.BacalhauConfig, error) {
-	return initConfig(path,
-		WithDefaultConfig(getDefaultConfig(path)),
-		WithFileHandler(ReadConfigHandler),
-	)
-}
+type Option = func(s *config)
 
-func getDefaultConfig(path string) types.BacalhauConfig {
-	// derive the default config for the specified environment.
-	defaultConfig := ForEnvironment()
-
-	// set default values for path dependent config.
-	defaultConfig.User.KeyPath = filepath.Join(path, UserPrivateKeyFileName)
-	defaultConfig.User.Libp2pKeyPath = filepath.Join(path, Libp2pPrivateKeyFileName)
-	defaultConfig.Node.ExecutorPluginPath = filepath.Join(path, PluginsPath)
-	defaultConfig.Node.ComputeStoragePath = filepath.Join(path, ComputeStoragesPath)
-	defaultConfig.Node.Compute.ExecutionStore.Path = filepath.Join(path, ComputeExecutionsStorePath)
-	defaultConfig.Node.Requester.JobStore.Path = filepath.Join(path, OrchestratorJobStorePath)
-	defaultConfig.Update.CheckStatePath = filepath.Join(path, UpdateCheckStatePath)
-	defaultConfig.Auth.TokensPath = filepath.Join(path, TokensPath)
-
-	// We default to the folder which contains the job store, and add
-	// a subfolder for the network store.
-	defaultConfig.Node.Network.StoreDir = filepath.Join(
-		filepath.Dir(defaultConfig.Node.Requester.JobStore.Path),
-		NetworkTransportStore,
-	)
-
-	return defaultConfig
+func WithDefaultConfig(cfg types.BacalhauConfig) Option {
+	return func(c *config) {
+		c.defaultCfg = cfg
+	}
 }
 
-type Params struct {
-	FileName      string
-	FileHandler   func(fileName string) error
-	DefaultConfig types.BacalhauConfig
+func New(opts ...Option) *config {
+	c := &config{
+		user:       viper.New(),
+		system:     viper.New(),
+		defaultCfg: configenv.Production,
+	}
+	for _, opt := range opts {
+		opt(c)
+	}
+	c.user.SetEnvPrefix(environmentVariablePrefix)
+	c.user.SetTypeByDefaultValue(inferConfigTypes)
+	c.user.AutomaticEnv()
+	c.user.SetEnvKeyReplacer(environmentVariableReplace)
+	c.setDefault(c.defaultCfg)
+	return c
 }
 
-func initConfig(path string, opts ...Option) (types.BacalhauConfig, error) {
-	params := &Params{
-		FileName:      ConfigFileName,
-		FileHandler:   NoopConfigHandler,
-		DefaultConfig: ForEnvironment(),
+func (c *config) RepoPath() (string, error) {
+	repoPath := c.system.GetString("repo")
+	if repoPath == "" {
+		return "", fmt.Errorf("repo path not configured")
 	}
+	return repoPath, nil
+}
 
-	for _, opt := range opts {
-		opt(params)
+func (c *config) Load(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		// if the config file doesn't exist then we obviously cannot load it
+		return fmt.Errorf("config file not found at at path: %q", path)
```

Comment:
Load now requires a config to be present at the path, if no file is there is returns an error.